Userland/Libraries/LibThreading/BackgroundAction.h

```diff
@@ -39,50 +40,80 @@ class BackgroundAction final : public Core::Object
     C_OBJECT(BackgroundAction);
 
 public:
-    void cancel()
-    {
-        m_cancelled = true;
-    }
-
-    bool is_cancelled() const
-    {
-        return m_cancelled;
-    }
+    // Promise is an implementation detail of BackgroundAction in order to communicate with EventLoop.
+    // All of the promise's callbacks and state are either managed by us or by EventLoop.
+    using Promise = Core::Promise<NonnullRefPtr<Core::Object>>;
 
     virtual ~BackgroundAction() = default;
 
+    Optional<Result> const& result() const { return m_result; }
+    Optional<Result>& result() { return m_result; }
+
+    void cancel() { m_cancelled = true; }
+    // If your action is long-running, you should periodically check the cancel state and possibly return early.
+    bool is_canceled() const { return m_cancelled; }
```

Comment:
there it is! a double »l«!

Reply:
Okay, I will fix that, but I really want to get this PR merged now, pretty please

Reply:
hahahh! thatʼs not on me! 😅
(feel free to merge that all the responsible people! 😄)